src/services/catalog.service.ts

```diff
@@ -138,6 +139,9 @@ export class CatalogService {
     let allSubCatOfOther: any = _.get(otherMainCategory, 'subCategories[0]');
 
     _.each(items, (item) => {
+      if (item.vendor) {
+        this.vendors.push(item.vendor);
+      }
```

Comment:
It's more efficient to use a map so you don't have to use `_.uniq`. I'm a bit worried about performance when the catalog grows large.

```js
let vendors = {};

// ...

if (item.vendor) {
  vendors[item.vendor] = true;
}

// ...

this.vendors = _.keys(vendors);
```